server/src/main/java/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -358,7 +358,7 @@ public TemplateProfile prepare(GetUploadParamsForTemplateCmd cmd) throws Resourc
                 BooleanUtils.toBoolean(cmd.isFeatured()), BooleanUtils.toBoolean(cmd.isExtractable()), cmd.getFormat(), osTypeId,
                 cmd.getZoneId(), HypervisorType.getType(cmd.getHypervisor()), cmd.getChecksum(),
                 cmd.getTemplateTag(), cmd.getEntityOwnerId(), cmd.getDetails(), BooleanUtils.toBoolean(cmd.isSshKeyEnabled()),
-                BooleanUtils.toBoolean(cmd.isDynamicallyScalable()), BooleanUtils.toBoolean(cmd.isRoutingType()));
+                BooleanUtils.toBoolean(cmd.isDynamicallyScalable()), BooleanUtils.toBoolean(cmd.isRoutingType()), cmd.isDeployAsIs());
         return prepareUploadParamsInternal(params);
```

Comment:
@nvazquez I think we have to pass params.isDeployAsIs() value in prepare() method inside prepareUploadParamsInternal() method. Otherwise I don't see the usage of deployAsIS from UploadParamsBase.

Reply:
@nvazquez at line 340

Reply:
Thanks guys, done. Somehow reverted it. Will re-test after this